libdevcore/CommonData.cpp

```diff
@@ -35,13 +35,26 @@
 using namespace std;
 using namespace dev;
 
-bool dev::isHex(string const& _s)
+namespace
 {
-	unsigned i = (_s.size() >= 2 && _s.substr(0, 2) == "0x") ? 2 : 0;
-	for (; i < _s.size(); ++i)
-		if (fromHex(_s[i], WhenError::DontThrow) == -1)
-			return false;
-	return true;
+int _fromHex(char _i) noexcept
```

Comment:
We could name it `fromHexNibble` / `fromHexChar` but we can also leave it like that.